lib/puma/server.rb

```diff
@@ -291,7 +291,7 @@ def handle_servers
 
                     pool << client
                     busy_threads = pool.wait_until_not_full
-                    if busy_threads == 0
```

Comment:
Well, the problem here is that if busy_threads != 0, it's not really out of band anymore, is it?